cmake/onnxruntime_optimizer.cmake

```diff
@@ -11,6 +11,21 @@ if (onnxruntime_MINIMAL_BUILD)
     "${ONNXRUNTIME_ROOT}/core/optimizer/graph_transformer.cc"
   )
 
+  if (onnxruntime_EXTENDED_MINIMAL_BUILD AND onnxruntime_USE_NNAPI_BUILTIN)
+    list(APPEND onnxruntime_optimizer_src_patterns
```

Comment:
Do we need a separate list or can we extend the one for onnxruntime_ENABLE_RUNTIME_OPTIMIZATION_REPLAY_IN_MINIMAL_BUILD? 

If you add QDQ handling to NNAPI it requires the handling for the CPU EP in case NNAPI isn't available.

Reply:
also as an aside and not for this PR: we should refine the name onnxruntime_ENABLE_RUNTIME_OPTIMIZATION_REPLAY_IN_MINIMAL_BUILD - I think it's too specific now. perhaps onnxruntime_ENABLE_RUNTIME_OPTIMIZATION_IN_MINIMAL_BUILD

Reply:
Will keep this way for now, and we can consolidate the compile script and ifdef(s) after most of the components is ready